test/test_nn.py

```diff
@@ -11392,9 +11391,159 @@ def __init__(self):
                 self.assertEqual(p.grad.to(devices[0]), pe.grad)
 
 
+instantiate_device_type_tests(TestNNDeviceType, globals())
```

Comment:
Leave this line near the end of the file, next to `instantiate_device_type_tests(TestConv2dExt, globals())`